pkg/cli/admin/release/extract.go

```diff
@@ -296,111 +299,64 @@ func (o *ExtractOptions) Run() error {
-						return true, nil
-					}
-					klog.V(4).Infof("Found manifest %s", hdr.Name)
-					raw, err := ioutil.ReadAll(r)
-					if err != nil {
-						manifestErrs = append(manifestErrs, errors.Wrapf(err, "error reading file %s", hdr.Name))
-						return true, nil
-					}
-					ms, err := manifest.ParseManifests(bytes.NewReader(raw))
-					if err != nil {
-						manifestErrs = append(manifestErrs, errors.Wrapf(err, "error parsing %s", hdr.Name))
-						return true, nil
-					}
-					o.Manifests = append(o.Manifests, ms...)
-				}
-				return true, nil
-			}
-		}
-		if err := opts.Run(); err != nil {
-			return err
-		}
-		if !found {
-			return fmt.Errorf("image did not contain %s", o.File)
-		}
-
-		// Only output manifest errors if manifests were being extracted.
-		// Do not return an error so current operation, e.g. mirroring, continues.
-		if o.ExtractManifests && len(manifestErrs) > 0 {
-			fmt.Fprintf(o.ErrOut, "Errors: %s\n", errorList(manifestErrs))
-		}
-		return nil
+	tarEntryCallbacks := []extract.TarEntryFunc{}
 
-	case o.CredentialsRequests:
+	var manifestErrs []error
+	if o.ExtractManifests {
 		expectedProviderSpecKind := ""
-		if len(o.Cloud) > 0 {
+		if o.Cloud != "" {
 			expectedProviderSpecKind = credRequestCloudProviderSpecKindMapping[o.Cloud]
 		}
-		opts.TarEntryCallback = func(hdr *tar.Header, _ extract.LayerInfo, r io.Reader) (bool, error) {
+
+		tarEntryCallbacks = append(tarEntryCallbacks, func(hdr *tar.Header, _ extract.LayerInfo, r io.Reader) (bool, error) {
 			if ext := path.Ext(hdr.Name); len(ext) == 0 || !(ext == ".yaml" || ext == ".yml" || ext == ".json") {
 				return true, nil
 			}
 			klog.V(4).Infof("Found manifest %s", hdr.Name)
-			raw, err := ioutil.ReadAll(r)
+			raw, err := io.ReadAll(r)
 			if err != nil {
-				return false, errors.Wrapf(err, "error reading file %s", hdr.Name)
+				manifestErrs = append(manifestErrs, errors.Wrapf(err, "error reading file %s", hdr.Name))
+				return true, nil
 			}
 			ms, err := manifest.ParseManifests(bytes.NewReader(raw))
 			if err != nil {
-				return false, errors.Wrapf(err, "error parsing %s", hdr.Name)
+				manifestErrs = append(manifestErrs, errors.Wrapf(err, "error parsing %s", hdr.Name))
+				return true, nil
 			}
-			credRequestManifests := []manifest.Manifest{}
+
+			o.Manifests = append(o.Manifests, ms...)
+
+			manifestsToWrite := make([]manifest.Manifest, 0, len(ms))
 			for _, m := range ms {
-				if m.GVK != credentialsRequestGVK {
-					continue
-				}
-				if len(expectedProviderSpecKind) > 0 {
-					kind, _, err := unstructured.NestedString(m.Obj.Object, "spec", "providerSpec", "kind")
-					if err != nil {
-						return false, errors.Wrap(err, "error extracting cred request kind")
-					}
-					if kind != expectedProviderSpecKind {
+				if o.CredentialsRequests {
+					if m.GVK != credentialsRequestGVK {
 						continue
 					}
+					if expectedProviderSpecKind != "" {
```

Comment:
maybe we could simplify the conditions in the loop and avoid multiple `continue`

```go
for _, m := range ms {
	if o.CredentialsRequests && m.GVK == credentialsRequestGVK && expectedProviderSpecKind != "" {
		kind, _, err := unstructured.NestedString(m.Obj.Object, "spec", "providerSpec", "kind")
		if err != nil {
			return false, errors.Wrap(err, "error extracting cred request kind")
		}
		if kind == expectedProviderSpecKind {
			manifestsToWrite = append(manifestsToWrite, m)
		}
	}
}
```

Reply:
I wanted to keep the `manifestsToWrite = append(manifestsToWrite, m)` outside of the `o.CredentialsRequests` branch, so I had a single location to add a `m.Include(...)` call in follow-up work for `--included`.